docs/in-depth.md

```diff
@@ -0,0 +1,220 @@
+A good way to understanding the sortition pool design is to understand the
+original problem and naive approach, and then understand the optimizations away
+from that naive solution.
+
+### Original Problem and Naive Approach
+
+Say that we have a list of operators like `[Alice, Bob, Carol, David]`, and
+each operator has a different amount of stake in the pool `{Alice: 20, Bob: 30:
+Carol: 40: David: 50}`. We need to select a group of operators (say, 3), where
```

Comment:
In the examples we use, the stake values have the ascending order. Maybe it would be good to add somewhere a disclaimer that this is not a requirement and algorithm works fine no matter what are the values?

Reply:
Also, in all the examples we use the number of operators is equal to some factor of `2`. We could add some note that if number of operators is between `2^(n-1)` and `2^n`, then we construct the tree like for `2^n` operators, but with `0`s in place of missing operators.

Reply:
I think I'll switch one of the stake values, but I actually think that talking about the various edge cases *confuses* the message rather than helps to build intuition. As in, the purpose of this document isn't to fully explain and spec out the sortition pool, but rather to give someone an intuitive foundation for the concepts.

I think this is especially the case in the for the binary tree, where that isn't the data structure we us, but rather one that only exists for illustration.

Reply:
https://github.com/keep-network/sortition-pools/pull/157/commits/b4180f4007cbe4aa2f21887dea5fc3057ea1fbf5